Release YOUTUBE Downloader v2.9

tormyvancool opened the pull request:
1.0 2024-26-10
    # First Release
1.1 2024-26-10
    + Processes Notifications
    - /Video/
    + /Videos/
1.2 2024-26-10
    - --merge-output-format mp4
    + -S vcodec:h264,res,acodec:aac
1.3 2024-26-10
    - 10
    + 2
1.4 2024-26-10
    - 2
    + 5
1.5 2024-26-10
    - 5
    + 1
    # Unified Update
1.6 2024-26-10 - 1
    + 2
    + Version
1.7 2024-27-10
    - 'start "" "' from all O.S.s
    + 'start "UPDATE & DOWNLOAD" "' Win
1.8 2024-27-10
    - GGGGG = ''
    - 1
    + Start = '"'
    + 2
1.9 2024-27-10
    + Check saved project
    - 1
    + 2
2.0 2024-27-10
    - "chmod +x " ..  MainPath
    + 'chmod +x "' ..  MainPath .. '"'
    # Ordered Variables
    - 2
    + 1
    + Apple Trial
2.3 2024-27-10
    # Linux execution correction
    + Credits
    # 2.1 and 2.2 just trials due issues with Linux and Apple
2.31 2024-28-10
    # Binaries directly form the source
2.32 2024-28-10
    - yt-dlp
    + yt-dlp_linux
2.4 2024-29-10
    # Adjusted header style for production
2.5 2024-11-04
    - Various 
    + VideoPath = 'Video'
2.6 2024-11-05
    + check for temrination of temporary file upfrotn import the video
2.7 2024-11-05
    - Check Routine
2.8 2024-11-06
    + Detects Nework Interruptions during download
    + Removes leftovers
    + URLs as filename: forbidden
    + Limitation to only alphanumerical characters
2.9 2024-11-06
    + Check IfFileExists: Overwrite, Newname, Exit

tormyvancool committed:
Release YOUTUBE Downloader v2.9

1.0 2024-26-10
    # First Release
1.1 2024-26-10
    + Processes Notifications
    - /Video/
    + /Videos/
1.2 2024-26-10
    - --merge-output-format mp4
    + -S vcodec:h264,res,acodec:aac
1.3 2024-26-10
    - 10
    + 2
1.4 2024-26-10
    - 2
    + 5
1.5 2024-26-10
    - 5
    + 1
    # Unified Update
1.6 2024-26-10 - 1
    + 2
    + Version
1.7 2024-27-10
    - 'start "" "' from all O.S.s
    + 'start "UPDATE & DOWNLOAD" "' Win
1.8 2024-27-10
    - GGGGG = ''
    - 1
    + Start = '"'
    + 2
1.9 2024-27-10
    + Check saved project
    - 1
    + 2
2.0 2024-27-10
    - "chmod +x " ..  MainPath
    + 'chmod +x "' ..  MainPath .. '"'
    # Ordered Variables
    - 2
    + 1
    + Apple Trial
2.3 2024-27-10
    # Linux execution correction
    + Credits
    # 2.1 and 2.2 just trials due issues with Linux and Apple
2.31 2024-28-10
    # Binaries directly form the source
2.32 2024-28-10
    - yt-dlp
    + yt-dlp_linux
2.4 2024-29-10
    # Adjusted header style for production
2.5 2024-11-04
    - Various 
    + VideoPath = 'Video'
2.6 2024-11-05
    + check for temrination of temporary file upfrotn import the video
2.7 2024-11-05
    - Check Routine
2.8 2024-11-06
    + Detects Nework Interruptions during download
    + Removes leftovers
    + URLs as filename: forbidden
    + Limitation to only alphanumerical characters
2.9 2024-11-06
    + Check IfFileExists: Overwrite, Newname, Exit

cfillion commented on Video/tormyvancool_YOUTUBE Downloader.lua:
```diff
@@ -0,0 +1,367 @@
+              ResDel = 'rm "' .. ProjDir .. '\\Videos\\*.mp4.part"'
+          elseif OpSys == 1 then
+              ResFiles = 'dir /b "' .. ProjDir .. '\\Videos\\*.mp4.part"'
+              ResDel = 'del "' .. ProjDir .. '\\Videos\\*.mp4.part"' 
```
There are API functions for this that don't require going through system commands. Also (at least) the Unix code path is not only dangerous but also broken: the star within quotes is interpreted literally not as a globbing operator. And no backslashes as directory separator on Unix.

tormyvancool replied:
I tried that command in Linux virtual machine (Ubuntu) as it is, before writing it into this  code and it works.
About the APIs you are referring to: which are?
Are they doing exactly what I want from these lines. The goal is to remove any `.mp4.part` file that's forming inside that specific subfolder.
If knew they existed I could use them; APIs are making my life easier (when and IF correctly explained)

NOTE "any protection against injection"???

cfillion replied:
> and it works.

It cannot for the two reasons listed above. If the temp files are removed it's by yt-dlp not by those two commands.

![Screen Shot 2024-11-08 at 05 47 37](https://github.com/user-attachments/assets/e22d133c-690e-42cf-ada3-3f033d8082ff)

> about the APIs you are referring to: which are?

`reaper.EnumerateFiles` and `os.remove`

> "any protection against injection"???

From the previous PR:

> Sure the macOS/Linux code paths (double-quoted w/o escaping) likely works for most cases. But it will fail for some users because the full range of valid inputs is not considered. And the failure mode (arbitrary command injection) allows for catastrophic outcomes (even if unlikely in practice).
>
> Input is any non-null bytes representing a path to an existing file. Output is a shell script to be interpreted by sh, bash or cmd.exe. All inputs must map to a valid output.
>
> Right now many inputs can make the shell behave unexpectedly (technically [arbitrary command injection](https://owasp.org/www-community/attacks/Command_Injection)).
>
> (cmd.exe has very different syntax rules than sh/bash! APIs such as os.execute are hard to get right and [widely](https://www.reddit.com/r/C_Programming/comments/uokyqw/comment/i8f5ot1/) [discouraged](https://wiki.sei.cmu.edu/confluence/pages/viewpage.action?pageId=87152177).)

tormyvancool replied:
yt-dlp does not remove these files self when the network fails. No way.
These commands are entering in actions ONLY if at the end of the process, yt-dlp failed/crashed and .mp4.part will still there forever as leftovers, otherwise.

Instead if yt-dlp works fine, they do not remove anything because there is nothing to be removed.

cfillion replied:
There's also no way `rm "/project/dir\Videos\*.mp4.part"` does anything beside output `No such file or directory` to stderr... (Unless there's a file literally named `Videos\*.mp4.part` with both the backslash and the star in the filename.)

tormyvancool replied:
I'm checking what I can do to get the identical thing I want, by the use of the suggested APIs

and for the "rm" not clue why it worked on my virtual machine then.

tormyvancool replied:
NOTE in `integer fileindex` it will be nice to understand what's expected. I just used the -1 only

I supposed it should be used in this way  `ResFiles = reaper.EnumerateFiles( ProjDir .. '/Videos/*.mp4.part', 1)`
But unlike the commands I wrote and they are working, this one doesn't :-/

cfillion replied:
`reaper.EnumerateFiles(ProjDir .. '/Videos', i)` in a loop with i incremented by one each iteration starting from 0 until nil is returned. Check if the returned filename ends with '.mp4.part' and delete if so...

tormyvancool replied:
Ok found the way

```
              a=0
              test = ""
              repeat
              reaper.ShowConsoleMsg(test)
              test = reaper.EnumerateFiles( ProjDir .. "\\Videos\\*.mp4.part", a)
              a = a + 1
              until(test == nil)
```

I will adapt it and implement it

tormyvancool replied:
Sorry I read you after have posted mien one. Ok indeed I'm doing that implementation and some part of the code mus be rewritten although it will be simplified and more efficient

tormyvancool replied:
Committed

cfillion commented on Video/tormyvancool_YOUTUBE Downloader.lua:
```diff
@@ -0,0 +1,367 @@
+-- @provides
+--   [windows] yt-dlp/yt-dlp.exe https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp.exe
+--   [linux] yt-dlp/yt-dlp_linux https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp
+--   [darwin] yt-dlp/yt-dlp_macos https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp_macos
```
Should un-list for unsupported 32-bit Windows/macOS, and for the script itself too so that it cannot be installed where it can't be used.

tormyvancool replied:
For the unsupported 32bits I will modify what's written between `[]` but how to avoid to download the script itself on 32bit machines: not any clue (I never did it).

EDIT/ on the ReaPack website I see now the platform dropdown is valid also for the script;
Should I just add the  ` -- @provides [win64] ` oly?
tu there is not an option for also MAC and Linux ... I f I chose oen I can't chose for the other ones as well.

How to make it downloadable only from 64bit users for ALL O.S.es?

cfillion replied:
Repeat the `provides` line for the script itself for each platform (dot = shortcut for the current file):

```lua
-- @provides
--   [win64]        .
--   [darwin64]     .
--   [darwin-arm64] .
--   [linux]        .
```

Using the package editor: click on the duplicate file button in the list on the left.

EDIT: Ah the editor does not detect this idiom and instead generates more verbose metadata (`[linux] . > author_Filename.lua`). It's functionally equivalent, just redundant (`>` is the rename operator). Will look into improving that...

tormyvancool replied:
Ok I will.

tormyvancool replied:
Committed

cfillion commented on Video/tormyvancool_YOUTUBE Downloader.lua:
```diff
@@ -0,0 +1,367 @@
+        end
+        if OS == "OSX32" or OS == "OSX64" or OS == "macOS-arm64" then
+          MainPath  = './yt-dlp_macos'
+          Start = 'cd "' .. CallPath .. '" && chmod +x ' .. dlpMac .. ' && '
+          os.execute('chmod +x "' ..  MainPath .. '"')
```
Same issue as last time here.

tormyvancool replied:
**If you are talking about the line 145**: it seems in Mac when you call directly the application (1 single command instead of all that "CDs") it returns that the file doesn't exists.

**If you are talking about line146**: On Linux I tried to give more than one time `chmod +x` everything works. So I don't see evidence to do it in a different way since when the yt-dlp is downloaded it seems do not have the execution flags. Only readable.

cfillion replied:
Line 146 only runs on macOS. It attempts to set the execute bit on yt-dlp_macos in whichever directory is current (unlikely to be yours the first time). So the line is effectively no-op. Only the one at line 145 is effective (ignoring the same injection problem). By the way it's checking `OS == "OSX32"` despite the script (well, the yt-dlp binary) not supporting macOS 32-bit?

tormyvancool replied:
> Line 146 only runs on macOS. It attempts to set the execute bit on yt-dlp_macos in whichever directory is current (unlikely to be yours the first time). So the line is effectively no-op. Only the one at line 145 is effective (ignoring the same injection problem). By the way it's checking `OS == "OSX32"` despite the script (well, the yt-dlp binary) not supporting macOS 32-bit?

Actually even the very first time is my directory.
It creates firs the yt-dlp subtir
then, on it, it will download the yt-dlp.something
Only from that moment on the flag is changed on +x

in can't be otherwise. or it means the script is not executed linearly form line 1 to the last and something si skipped.
But if so, it means something else is not working as it should.
So that "cd " can't be wrong

NOTE: I solved the checking for OSX2 that however it didn't do anything wrong if there.
I have always the same if then else end on a separate file and I copy and paste it.
If the script is installed on any machine: ti works.
And that check is not damaging anything.

cfillion replied:
> It creates firs the yt-dlp subtir
> then, on it, it will download the yt-dlp.something
>
> in can't be otherwise

That's not in the script's code? `getOS` is called from line 175 and it's the first thing that's done after clearing the console and defining a few variables and functions. Nothing changes the current working directory of REAPER's process.

The `yt-dlp` directory is created and filled when the package is installed by ReaPack. Not when running the script. And ReaPack does so without messing with REAPER's working dir.

tormyvancool replied:
> > It creates firs the yt-dlp subtir
> > then, on it, it will download the yt-dlp.something
> > in can't be otherwise
> 
> That's not in the script's code? `getOS` is called from line 175 and it's the first thing that's done after clearing the console and defining a few variables and functions. Nothing changes the current working directory of REAPER's process.
> 
> The `yt-dlp` directory is created and filled when the package is installed by ReaPack. Not when running the script. And ReaPack does so without messing with REAPER's working dir.

yes ReaPack does it because the code is telling to do so, due the lines @provide.
Hence the `yt-dlp` subdir is created, and the script behaves in the way I said.

EDIT: instead if you want the script goes to check if really that subdir is created otherwise alerts and stop running, is something else. But I have not clue (now)how to access the scripts dir to investigate it's existence with Reaper's APIs.

tormyvancool replied:
I need your feedback to work on this.

tormyvancool replied:
ok no feedback = free interpretation :-)
It means: I found the way to check if that subfolder is created or not.
If not created a message will appear telling that the script MUST be installed via reapack repository.

Once implemented and tested I will commit it.

tormyvancool replied:
There is again to work on the `getOS()` function I see. I will work on the path now and I will update the code.

tormyvancool replied:
Ok super. Committed. the path is detected by the Reaper's APIs and it's not more hardencoded

cfillion replied:
> EDIT: instead if you want the script goes to check if really that subdir

I didn't say that? There is no reason to check whether ReaPack has installed yt-dlp. It either does or the entire package is not installed at all & the user is informed of the error.

I was talking about the broken & redundant chmod which is still not removed.

cfillion reviewed (commented):
It's an improvement over last time however there are still significant design flaws that I can't accept on a default repository. Most notably the complete absence of any protection against injection which is unchanged from the previous PR.

In the current state I think it would be better released on your own repository.

tormyvancool committed:
Update tormyvancool_YOUTUBE Downloader.lua

Download admittend only on supported O.S.es

tormyvancool committed:
Update tormyvancool_YOUTUBE Downloader.lua

Implemented OS independent code to check for leftovers and remove them

tormyvancool committed:
Update tormyvancool_YOUTUBE Downloader.lua

As discussed here
https://github.com/ReaTeam/ReaScripts/pull/1457#discussion_r1834571785

tormyvancool commented:
> It's an improvement over last time however there are still significant design flaws that I can't accept on a default repository. Most notably the complete absence of any protection against injection which is unchanged from the previous PR.
> 
> In the current state I think it would be better released on your own repository.

I went through all what highlighted and committed the solutions.

tormyvancool committed:
Update tormyvancool_YOUTUBE Downloader.lua

Fixed the function getOS()

tormyvancool committed:
Update tormyvancool_YOUTUBE Downloader.lua

Implemented the solution here, to detect the script's folder
https://forums.cockos.com/showpost.php?p=2821129&postcount=11

tormyvancool committed:
Update tormyvancool_YOUTUBE Downloader.lua

Added a check:
if the supposed downloaded file doesn't exists OR it is 0 bytes sized, the script informs the operator and just by clicking OG ends up.
No empty track - or track with 0 bytes file - will be created in Reaper

tormyvancool commented:
Ok I think i've done with this script. All what we discussed and also something more, was done and the "something more" was added. 
So in case something goes wrong, the script shouldn't freeze Reaper indefinitely, and informs the operator stopping itself and let Reaper free of working.

cfillion commented:
Going to reject this for now. Thanks for applying some of the smaller suggestions, however the major flaws pointed out in both this and the previous PRs are still not addressed. Shell command execution requires more care than most scripts and must either be done right or not attempted.

tormyvancool commented:
> Shell command execution requires more care than most scripts and must either be done right or not attempted.

Shell commands are for what it should be done the way I expect. The function you suggested me: doesn't and it creates more issues than benefits.

> I didn't say that? There is no reason to check whether ReaPack has installed yt-dlp. It either does or the entire package is not > installed at all & the user is informed of the error.
>
> I was talking about the broken & redundant chmod which is still not removed.

That's why I asked for the confirmation of what I understood. You came today. I asked 3 days ago. And I added "ok no feedback = free interpretation :-)"  so meanwhile I worked on what I understood.

The redundant `chmod +x` is required since during the installation of the package it seems there is not way to guarantee the `chmod +x `as required to the downloaded binary. And this is was clearly issued during the testing of the previous versions. Indeed yt-dlp was downloaded without that flag (it turned out to me only read mode). Solution: run that command to be sure is +x 

If I ask to the O.S. to "chmod +x" when the file already is in "x" nothing happens. It survives on this and it doesn't require more than some microsecond.

tormyvancool commented:
NOTE the fact you closed this PR, obliges me to pass again that uncomfortable form.It requires more effort there tan to remove a redundancy on os.execute() with the issues on downloading only the 64 bits etc ...

cfillion commented:
> Shell commands are for what it should be done the way I expect. The function you suggested me: doesn't and it creates more issues than benefits.

`os.execute` is fine if the input is properly sanitized ([not](https://daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULESDOC) [trivial](https://flatt.tech/research/posts/batbadbut-you-cant-securely-execute-commands-on-windows/) on Windows, not too bad on Unix-likes). `ExecProcess` is easier in this aspect, but of course it comes with limitations (no escape character = worse than `os.execute` for compatibility) and can be tricky to use without understanding it.

> That's why I asked for the confirmation of what I understood. You came today. I asked 3 days ago.

...are you really complaining because I didn't reply within the 2/6 hours between your "I need your feedback to work on this" and "ok no feedback = free interpretation" / "Ok super. Committed." replies?

> The redundant chmod +x is required

Two chmod are redundant. One is enough. Looks like you tried to fix that in #1461 by removing one of them. Good but the one that's left is not going to work without being provided the correct path to the file.

> NOTE the fact you closed this PR, obliges me to pass [...]

You can re-create a PR without going through the form again (same git branch). But please go ahead and release it in your own repository instead. I'm sure the script is useful and mostly works but I can't accept it here (where it's going to be shipped to all users) unless the various packaging and core functionality issues are fully addressed.

- Hardened against command injection
- `chmod` not broken on macOS – since you're testing on Linux: macOS is pretty much the same yet your code is different for some reason
- `yt-dlp` pinned to a specific version (when an update –yours or yt-dlp– breaks something, users need to trust ReaPack's downgrade feature to undo the update without breaking a sweat)
- No multi-version changelog

While at it (optional, out of scope for a packaging review) the code quality could be improved. There's unused leftovers, inconsistencies and typos (such "Downlaod" and "filname")...

tormyvancool commented:
> > Shell commands are for what it should be done the way I expect. The function you suggested me: doesn't and it creates more issues than benefits.
> 
> `os.execute` is fine if the input is properly sanitized ([not](https://daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULESDOC) [trivial](https://flatt.tech/research/posts/batbadbut-you-cant-securely-execute-commands-on-windows/) on Windows, not too bad on Unix-likes). `ExecProcess` is easier in this aspect, but of course it comes with limitations (no escape character = worse than `os.execute` for compatibility) and can be tricky to use without understanding it.

How to sanitize it more? I have not clue. Or I wasn't make mistakes.
If you notice that os.execute() base don OSes in the beginning are just paths taken by REPAER's APIs.
The same for the variables "Update" and "Video" used into the os.execute() almost at the end of the script.

The input fields (the only oen way I see a potential injection)

- the first one verifies is an URL and it must be valid
- the second one has limited inputs: only alphanumerically characters.

> ...are you really complaining because I didn't reply within the 2/6 hours between your "I need your feedback to work on this" and "ok no feedback = free interpretation" / "Ok super. Committed." replies?

In 3 days (not 6 ours) I was working on it trying also to better understand what you were saying. Do not have feedback I just applied what I did understand. It's not an accusation but it clears out how I was working.
however the control I have pt in place does its job. So ppl will install the script via ReaPack system avoiding to download just the script, copy and paste it and use it, missing the important settings.
 
> Two chmod are redundant. One is enough. Looks like you tried to fix that in #1461 by removing one of them. Good but the one that's left is not going to work without being provided the correct path to the file.

Yes I spotted it after this. That's why I corrected it and opened the PR that you closed now.
 
> You can re-create a PR without going through the form again (same git branch). But please go ahead and release it in your own repository instead. I'm sure the script is useful and mostly works but I can't accept it here (where it's going to be shipped to all users) unless the various packaging and core functionality issues are fully addressed.

That's why I started to ask for HELP everywhere (for the os.execute()). But I don't get the hints I do need to sanitize that damn os.execute()
I'm not born with the knowledge embedded or I wasn't here neither.

Example? tons; For instance how can I reopen a PR by updating the code avodigin that form and avoiding to open a fork?
I cannot edit the code pasting the new one now that's closed.

> * Hardened against command injection

This where I'm still asking for but not answers that really helps me not only to understand but to avoid future mistakes.

> * `chmod` not broken on macOS – since you're testing on Linux: macOS is pretty much the same yet your code is different for some reason

In the new code (3.0) I removed it.

> * `yt-dlp` pinned to a specific version (when an update –yours or yt-dlp– breaks something, users need to trust ReaPack's downgrade feature to undo the update without breaking a sweat)

yt-dlp is not released if it breaks everything. yes can have bugs like Reaper has bugs. But it's not breaking anything.

> * No multi-version changelog

I explained you that the detailed changelog served to keep history on what it was done and ReaPack has not that history since from 1.0 to 30 was published on my repository. The form on RePack site, is a text box that enables to paste the whole history.

> While at it (optional, out of scope for a packaging review) the code quality could be improved. There's unused leftovers, inconsistencies and typos (such "Downlaod" and "filname")...

cfillion commented:
> How to sanitize it more? I have not clue.

- https://www.gnu.org/software/bash/manual/bash.html#Quoting
- https://daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULESDOC
- https://flatt.tech/research/posts/batbadbut-you-cant-securely-execute-commands-on-windows/

> That's why I corrected it and opened the PR that you closed now.

You removed the chmod that was working and kept the one that didn't.

> yt-dlp is not released if it breaks everything. yes can have bugs like Reaper has bugs. But it's not breaking anything.

When something break, which also includes undesired changes of any kind, users need to be able to easily and reliably downgrade.

https://xkcd.com/1172/

> how can I reopen a PR

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

> I explained you that the detailed changelog served to keep history

The purpose of changelogs is to inform/warn users of changes between versions when they install or update. There is no purpose for changelogs before the first ReaPack release.

Keeping full development history is git's purpose.

> The form on RePack site, is a text box that enables to paste the whole history.

It says immediately below it: "Changelog for the current version only".

tormyvancool commented:
> > How to sanitize it more? I have not clue.
> 
> * https://www.gnu.org/software/bash/manual/bash.html#Quoting
> * https://daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULESDOC
> * https://flatt.tech/research/posts/batbadbut-you-cant-securely-execute-commands-on-windows/
> 
> > That's why I corrected it and opened the PR that you closed now.
> 
> You removed the chmod that was working and kept the one that didn't.

oops restored and removed the one inside the os.execute()

However about the quoting: what's need to be quoted? Since practically everything is generated by the REAPER's APIs
Or do you mean to avoid the double -quotes? For instance

from `MainPath = '"' .. CallPath .. dlpLnx .. '"'`
to `MainPath = "'" .. CallPath .. dlpLnx .. "'"` ?

On the pull request: once it's closed I have this
![image](https://github.com/user-attachments/assets/e0300c39-b01f-46bc-bd2b-6ebab6560fed)
I can't edit any longer.

An this
"In the "Branch" menu, choose the branch that contains your commits." is not valid once the PR is closed.

cfillion commented:
> restored and removed the one inside the os.execute()

👍 (Alternatively could have fixed the os.execute one by giving it the correct path.)

> However about the quoting: what's need to be quoted? 

(Linux&macOS) Single quotes are easier but the possibility of those variables containing single quotes and backslashes themselves has to be handled.

> once it's closed I have this

The grayed out 'edit file' button in the screenshot is a shortcut to a similar button on your fork.

Good luck with your script.

tormyvancool commented:
> > restored and removed the one inside the os.execute()
> 
> 👍 (Alternatively could have fixed the os.execute one by giving it the correct path.)
> 
> > However about the quoting: what's need to be quoted?
> 
> (Linux&macOS) Single quotes are easier but the possibility of those variables containing single quotes and backslashes themselves has to be handled.
> 
> > once it's closed I have this
> 
> The grayed out 'edit file' button in the screenshot is a shortcut to a similar button on your fork.
> 
> Good luck with your script.

The correct path in the last version was given by the API by Reaper. Indeed I changed the way these variables were generated.

About the quotes: besides the only few quotes I put, all the other parameters (like the path and filenames) are generated by Reaper's APIs. I think these APIs are already managing this issue and sanitizing the content.
